### DISCUSSION TITLE
Translation program

### DISCUSSION THREAD

**Discussion by alejandroRbit** [Magickbase/shaping#50] [Ideas]
CKB is universal, decentralized, open to everyone, and therefore inclusive. Following this ethos, accessibility to resources and info in different languages should be a high priority, which also has great advantages:

1. **Promotes decentralization.**
2. **Automation of translations:** Relieves the workload of maintainers.
3. **Ease of use:** Community members do not have to learn Github in depth and it is very intuitive.
4. **Encourages community participation:** Being easy to use allows and encourages the community to contribute to the project.
5. **Promote the feeling of belonging to the community - Culture:** By contributing to the project, community members feel that they are part of it, they feel that they can be a force for change, which makes them get involved and contribute more, creating a virtuous circle.
6. **Better UX:** Users can access network resources and applications in their native languages, with which they feel more comfortable.

Below are some alternatives tested and used successfully in other networks that can serve as a model for inspiration:

- **Bitcoin Core** uses **Transifex**.

https://explore.transifex.com/bitcoin/bitcoin/

Other projects that use Transifex are Litecoin or Dash:

For example, Dash translation process guide:
https://github.com/dashpay/dash/blob/master/doc/translation_process.md)

- **Ethereum** uses **Crowdin** https://crowdin.com/profile/ethdotorg

Some resources on the Ethereum model:

Ethereum Translation Program: https://ethereum.org/en/contributing/translation-program/
Ethereum Translation Style Guide: https://ethereum.org/en/contributing/translation-program/translators-guide/

- **Monero** uses **Weblate** https://translate.getmonero.org/

Some resources on the Monero model:

Translation tips for Monero translators: https://github.com/monero-ecosystem/monero-translations/blob/master/translation-tips.md
Step by step guide for Weblate: https://github.com/monero-ecosystem/monero-translations/blob/master/weblate.md

**Another option** would be not to use any of these platforms, doing it directly using Github. In any case, we should make a guide on how to contribute and good practices, among others. In my opinion, I think the best thing to do is to choose one of these options (or propose others) first, and from there start creating specific documentation for the translation program.

**Where it can be implemented:**

- Web page
- Articles
- Neuron
- Nexus
- Explorers
- RFCs
- Any type of documentation, application or web.
- Others.

**Acknowledgments and rewards for translators and contributors**

- Reward contributors with POAPs, NFTs (JoyID-CoTA, NFTNation Marketplace), SubDID (.bit), etc.
- Leaderboard.
- Certificates.
- Merch
- Event tickets

https://ethereum.org/en/contributing/translation-program/acknowledgements/

Eliminating barriers to entry we get the community to contribute more and better (previous points), but also through these incentives we get to involve it even more by rewarding it socially.

**Comment by LaloGarza** [comment]
I totally support this idea! And, can confirm that the advantage points mentioned does empowers a more diverse ecosystem.
Back in the day when I worked for Nervos I proposed using Lokalise or Crowdin for translation campaigns; I have helped launched a couple of translations programs and done translation work myself, using both translations platforms mentioned and as well directly on Github, and I found out that community is more open to contribute with an easy tool to do so [aka: any translation services platforms], and it is easier to coordinate efforts as well. 
Great to see this being proposed Alejandro 💪

**Comment by Keith-CY** [comment]
I've used **Crowdin** before, and it's very open to the community.
Now I prefer to integrate a ChatGPT translator into the development workflow(triggered by continuous integration/CI) and request feedback from the community.

With the help of ChatGPT CI, we can focus on the collaboration on GitHub.

**Reply by alejandroRbit** [reply]
Thanks Keith for your response. I think it's a good idea. ChatGPT's workflow is very similar to the current way of translating, which is mainly based on revision. Chema and I, for the Spanish translation of the web and all the articles, what we did was pass the texts through a translator like Deepl and then we reviewed the translation, fortunately the automatic translations between Spanish and English are quite complete. The biggest challenge in our case is regarding the more technical vocabulary, it is normally what consumes the most time and we always have the doubt of whether it is better to translate it or leave it original in English, that is why I found the Monero glossary very interesting: [https://github.com/monero-ecosystem/monero-translations/tree/master/terminology-guides](//github.com/monero-ecosystem/monero-translations/tree/master/terminology-guides). The other problem has been Github, since for the uninitiated it can be intimidating and not very intuitive (I say this from my own experience, I'm still learning), which discourages collaboration. It must be taken into account that there are members of the community who are willing to translate/review, but the medium/platform to implement the translations is a barrier, be it Github or another platform, hence the need for guidelanes or any other measure to reduce the difficulty and encourage participation (and reduce the workload for all of you too). Hopefully it serves as feedback.

**Comment by matt-nervos** [comment]
thanks Alejandro for compiling these resources. There is a lot of information about best practices and tips in here.

On the subject of platform, it looks like we can integrate Github with CrowdIn https://support.crowdin.com/github-integration/ and get the best of both worlds.

**Comment by Keith-CY** [comment]
Do you have any thoughts on how to collaborate with the community on internationalization? Once we reach a consensus, we can promote it to other teams and adopt a unified approach to reduce the barriers to supporting i18n. @Danie0918 @Sven-TBD @yanguoyu @homura @WhiteMinds

**Reply by poor-defined** [reply]
We could collaborate on some translation management system, like Weblate. It's open source.  The translation flow should also be taken into consideration while choosing the TMS.

**Reply by Keith-CY** [reply]
**Weblate** was mentioned at https://github.com/orgs/Magickbase/discussions/50#discussion-5215843
And **Transifex**, **Crowdin**, **ChatGPT** are also mentioned above

Would you have any thoughts on which one to go?

We hope the system is smart enough to recognize various formats of articles, especially the markdown.

I've recommended integrating **ChatGPT** with CI, but a drawback that a community member revealed is that **ChatGPT cannot analyze title/header/paragraph very well**. So the format of translation is malformed.

I believe the drawback will be/has been fixed by ChatGPT's thriving community. If so, I will still vote up to **Translate with ChatGPT with a review from community** because it does have a promising solution.

**Comment by homura** [comment]
I've never used a localization management platform before this discussion, so I may not be able to give constructive suggestions

I had a brief experience with Transifex and Crowdin, and I personally prefer Crowdin, because I feel it is more open. Crowdin has an app store like integration system. It provides support for glossary management, AI integration, multiple translation platform integration (e.g. DeepL), etc., and is easy to use

<img width="300" src="https://github.com/Magickbase/shaping/assets/7511174/81c6eb4e-6e04-4a2d-ad13-517a5c8db83f" />

<img width="600" src="https://github.com/Magickbase/shaping/assets/7511174/81e859a7-6a9f-4964-8483-d5c3874755fd" />
